rootston/seat.c

```diff
@@ -762,6 +768,14 @@ void roots_seat_set_focus(struct roots_seat *seat, struct roots_view *view) {
 		wlr_seat_keyboard_notify_enter(seat->seat, view->wlr_surface,
 			NULL, 0, NULL);
 	}
+
+	struct timespec now;
+	clock_gettime(CLOCK_MONOTONIC, &now);
+	struct roots_seat *all_seat;
+	wl_list_for_each(all_seat, &seat->input->seats, link) {
```

Comment:
This feels a little out of place here.
It's fitting to call it during `roots_seat_set_focus`, but it's logically part of the first few lines in the function (bring to top) and should probably be closer to that (maybe in a dedicated function?)

Reply:
I added it at the end because we do the same with keyboard enter just above, but happy to move it up - it only needs to be below `wl_list_insert(&seat->input->server->desktop->views, &view->link)`

Actually now that I look at it again it needs to be just after that and before the next prev = view check, because this can bubble a view that was already focused up.
I'll put this in its own function and move it there after work, I'm laaaate :D